controllers/vspheremachine_controller.go

```diff
@@ -340,7 +354,7 @@ func (r machineReconciler) reconcileNormal(ctx *context.MachineContext) (reconci
 				"unexpected error while reconciling provider ID for %s", ctx)
 		}
 		ctx.Logger.Info("provider ID is not reconciled")
-		return reconcile.Result{}, nil
+		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
```

Comment:
This is a hack and should be avoided long-term. Please add a `// TODO` to indicate this should be revisited.

Reply:
If we're not able to read the providerID from bios-uuid we'd have to wait for the resync period of 10min to reconcile this.